src/main/java/LeaveGroup/LeaveGroupInteractor.java

```diff
@@ -0,0 +1,68 @@
+package LeaveGroup;
+
+import Entities.Group;
+import Entities.User;
+
+import java.util.Objects;
+
+/**
+ * The leaveGroup use case.
+ * The user can only leave groups that they are currently a member of
+ * in which they are NOT the group leader.
+ */
+
+public class LeaveGroupInteractor implements LeaveGroupInputBoundary {
+
+    final LeaveGroupDsGateway dsGateway;
+    final LeaveGroupOutputBoundary outputBoundary;
+
+    /**
+     * The interactor for the leaveGroup use case.
+     * @param dsGateway the data access interface
+     * @param outputBoundary the output boundary implemented by the leaveGroupPresenter
+     */
+    public LeaveGroupInteractor(LeaveGroupDsGateway dsGateway,
+                                LeaveGroupOutputBoundary outputBoundary) {
+
+        this.dsGateway = dsGateway;
+        this.outputBoundary = outputBoundary;
+    }
+
+    /**
+     * @param requestModel the requestModel for the leaveGroup use case
+     * @return a successView or failView depending on whether the user has left the group
+     */
+    @Override
+    public LeaveGroupResponseModel leaveGroup(LeaveGroupRequestModel requestModel) {
+
+        if (!dsGateway.groupExists(requestModel.getGroupname())) {
+            return outputBoundary.prepareFailureView("Group does not exist.");
+        }
+
+        User user = dsGateway.getUser(requestModel.getUsername());
```

Comment:
I like how you included error messages for each error. It will be easier for the user to know what they did wrong, and/or when you are testing the code.